src/components/ShoppingList/ShoppingList.js

```diff
@@ -62,6 +62,90 @@ function ShoppingList({ listId }) {
+      ).plus({ days: item.purchaseInterval });
+    } else if (item.createdAt?.seconds) {
+      // if there's no purchase history, estimate it will be bought [purchaseInterval] days from when item was created
+      // (user provides this info at item creation)
+      nextPurchaseDate = DateTime.fromSeconds(item.createdAt.seconds).plus({
+        days: item.purchaseInterval,
+      });
+    } else return null;
+
+    const currentDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+    const daysRemaining = nextPurchaseDate.diff(currentDate, ['days']);
+    return Math.round(daysRemaining.as('days'));
+  };
+
+  const isItemInactive = (item) => {
+    const currentDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+
+    if (item.lastPurchaseDate?.seconds) {
+      // if an item has been purchased before, compare the time elapsed since lastPurchaseDate to the purchaseInterval
+      return (
+        currentDate
+          .diff(DateTime.fromSeconds(item.lastPurchaseDate.seconds), ['days'])
+          .as('days') >=
+        2 * item.purchaseInterval
+      );
+    } else if (item.createdAt?.seconds) {
+      // if an item has never been purchased, do the same calculating using createdAt date instead of lastPurchaseDate
+      return (
+        currentDate
+          .diff(DateTime.fromSeconds(item.createdAt.seconds), ['days'])
+          .as('days') >=
+        2 * item.purchaseInterval
+      );
+    } else return null;
+  };
+
+  const sortListItems = (itemA, itemB) => {
+    // if items being compared are both inactive, or neither is inactive, sort by days until next purchase
+    if (
+      (itemA.status === 'inactive' && itemB.status === 'inactive') ||
+      (itemA.status !== 'inactive' && itemB.status !== 'inactive')
+    ) {
+      if (itemA.daysToPurchase < itemB.daysToPurchase) return -1;
+      if (itemA.daysToPurchase > itemB.daysToPurchase) return 1;
+
+      // if we've made it this far, days to purchase is the same for both items so alphabetize
+      return itemA.itemName.localeCompare(itemB.itemName, 'en', {
+        sensitivity: 'base',
+        ignorePunctuation: true,
+      });
+    } else {
+      // if one item is inactive and the other is not, bump down the inactive item
+      if (itemA.status === 'inactive') return 1;
+      if (itemB.status === 'inactive') return -1;
+    }
+  };
+
+  const getItemStatus = (item) => {
+    if (isItemInactive(item)) return 'inactive';
+    if (item.daysToPurchase < 7) return 'soon';
+    if (item.daysToPurchase >= 7 && item.daysToPurchase <= 30)
+      return 'kind-of-soon';
+    if (item.daysToPurchase > 30) return 'not-soon';
+  };
+
+  // format the data from firestore into a sorted, filtered array of items
+  // add new properties: item.status and item.daysToPurchase
+  const itemsToDisplay = listItems?.docs
+    .filter((doc) => new RegExp(filter, 'i').test(doc.data().itemName))
+    .map((doc) => {
+      const item = doc.data();
+      item.id = doc.id;
```

Comment:
Thanks for making a single item since Firestore keeps these separate! This definitely makes it easier for all of us in the long-term

Reply:
I think so, and it seemed the easiest way to pass our results of getdaysToPurchase and getItemStatus to the ShoppingListItem component without needing to calculate them more than once (since we need them in this component for sorting).